app/layer_generators/main.go

```diff
@@ -218,6 +218,17 @@ func generateLayer(name, templateFile string) ([]byte, error) {
 			}
 			return ""
 		},
+		"shouldTrace": func(params map[string]bool, param string) string {
+			if _, ok := params[param]; ok {
+				return fmt.Sprintf(`span.SetTag("%s", %s)`, param, param)
+			}
+			for pName := range params {
```

Comment:
Maybe add a comment here to explain that param`.`field can be used to extract struct fields.